### ISSUE TITLE
Vastly increased incremental compilation times

### ISSUE THREAD

**Issue by jarohen** [jarohen/flow#4]
Incremental compilation times (using lein-cljsbuild) are up to about 10s for a moderately-sized app. This problem didn't exist in 0.2.0-b2 but does in 0.2.0-b4 (and also didn't crop up while I was developing it)

J

**Comment by jarohen** [comment]
So this only happens in whitespace mode (with optimisations it's quick as a flash) 

The time is nearly all taken on https://github.com/clojure/clojurescript/blob/9fd6bf5bd55421c3d5becacc5230ed661d6fb3c3/src/clj/cljs/closure.clj#L596, when the code is passed to the Closure compiler for the whitespace optimisation.

Trying a couple of approaches at the moment:
- Why is the Closure compiler taking so long?! Haven't got a long way into this - it's pretty gnarly and spectacularly Java-y.
- The code that Flow generates is quite nested with a lot of functional closures, and hence probably a lot for a compiler to keep in mind (although maybe not a whitespace compiler) - is there a way of making it less nested?
- Can I use 'none' optimisations in a sane way?
- Would it be possible to patch ClojureScript to output all the JS in one file but not optimise?

**Comment by jarohen** [comment]
Gone for option 3 - using 'none' optimisations in a sane way. Option 4 would be a lovely addition, if I/someone gets time.

I put some thought into 2 - the way Flow used to be architected (in particular, the `:declarations` part of the compiler) did get around a lot of these problems. It's doable under the new Flow implementation, but given I've got 'none' optimisations working, probably not that worthwhile.

Closing for now, further thoughts to come later!

Let me know below if this is a big impact - happy to pick up again if it affects others' usage.

James
